tests/derives.rs

```diff
@@ -0,0 +1,27 @@
+#[test]
+#[cfg(feature = "macros")]
+fn encode() {
+    use sqlx::encode::Encode;
+
+    #[derive(Encode)]
+    struct Foo(i32);
+
+    #[cfg(feature = "postgres")]
+    let _: Box<dyn Encode<sqlx::Postgres>> = Box::new(Foo(1));
```

Comment:
I would lift the type definition to the top level and then have `cfg`'d tests for MySQL and Postgres:

```rust
use sqlx::{decode::Decode, encode::Encode};

#[derive(Decode, Encode)]
struct Foo(i32);

#[cfg(feature = "postgres")]
fn encode_postgres() {
    ...
}
/// etc.
```

I would like to see tests that assert that `Encode<MySQL | Postgres>::encode()` produces the same result for `Foo` as it does for `i32` (using a value that fills all 4 bytes, like `0x11223344`).

Likewise for `Decode`, I would `Encode` the same value as `i32` for the given database and then assert that `<Foo as Decode<MySQL | Postgres>>` works and returns the same value internally.